### PR TITLE
Deprecate utility components

### PR DESCRIPTION
**Pull request by VanAnderson**
Deprecate Utility Components in favor of Box.

Closes https://github.com/github/primer/issues/223

### Merge checklist
- [x] Added/updated tests
- [x] Added/updated documentation
- [x] Tested in Chrome
- [x] Tested in Firefox
- [x] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/components/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: .changeset/metal-swans-allow.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@primer/components": minor
+---
+
+Deprecate utility components.
```
Comment (colebemis) [inline .changeset/metal-swans-allow.md:5]: It might be helpful for consumers if we listed all the deprecated components explicitly here.

**Review by colebemis — APPROVED** [review]
Excellent work, @VanAnderson! Thanks for going above and beyond by writing a codemod for these changes.

This looks good to go. But let's wait to merge this until https://github.com/primer/components/pull/1315 is merged.

File: src/stories/useFocusTrap.stories.tsx [hunk 2]
```diff
@@ -63,22 +63,22 @@ export const FocusTrap = () => {
   return (
     <>
       <HelperGlobalStyling />
-      <Flex flexDirection="column" alignItems="flex-start">
+      <Box display="flex" flexDirection="column" alignItems="flex-start">
         <MarginButton>Apple</MarginButton>
         <MarginButton>Banana</MarginButton>
         <MarginButton>Cantaloupe</MarginButton>
         <BorderBox borderColor="gray.5" ref={containerRef as React.RefObject<HTMLDivElement>} m={4} p={4}>
```
Comment (colebemis) [inline src/stories/useFocusTrap.stories.tsx:70]: Let's deprecate `BorderBox` in this PR as well.

File: src/Flex.tsx [hunk 3]
```diff
@@ -3,6 +3,9 @@ import Box, {BoxProps} from './Box'
 
 export type FlexProps = BoxProps
 
+/**
+ * @deprecated Please use the Box component instead.
```
Comment (colebemis) [inline src/Flex.tsx:7]: Can we provide more specific instructions on how to migrate to the Box component in these deprecation comments?
```suggestion
 * @deprecated Use the Box component instead (i.e. <Flex> → <Box display="flex">)
```

File: .changeset/metal-swans-allow.md (hunk 1 above)
Comment (colebemis) [reply .changeset/metal-swans-allow.md:5]: In addition to a list of deprecated components, let's also include instructions on how to run the codemod here.

File: src/Grid.tsx [hunk 4]
```diff
@@ -3,6 +3,9 @@ import Box, {BoxProps} from './Box'
 
 export type GridProps = BoxProps
 
+/**
+ * @deprecated Use the Box component instead (i.e. <Grid> → <Grid display="grid">)
```
Comment (colebemis) [inline src/Grid.tsx:7]:
```suggestion
 * @deprecated Use the Box component instead (i.e. <Grid> → <Box display="grid">)
```

Comment (VanAnderson) [reply src/Grid.tsx:7]: Ha, whoops!

File: src/Position.tsx [hunk 5]
```diff
@@ -5,34 +5,57 @@ import {ComponentProps} from './utils/types'
 
 type StyledPositionProps = {as?: React.ElementType}
 
+/**
+ * @deprecated Use the Box component instead (i.e. <Position> → <Box position="absolute">)
```
Comment (colebemis) [inline src/Position.tsx:9]:
```suggestion
 * @deprecated Use the Box component instead (i.e. <Position> → <Box>)
```

File: src/Popover.tsx [hunk 6]
```diff
@@ -43,7 +43,7 @@ const Popover = styled.div.attrs<StyledPopoverProps>(({className, caret}) => {
   ${sx};
 `
 
-const PopoverContent = styled(BorderBox)`
+const PopoverContent = styled(Box)`
   position: relative;
```
Comment (colebemis) [inline src/Popover.tsx:47]:
```suggestion
  border: 1px solid ${get('colors.border.primary')};
  border-radius: ${get('radii.2')};
  position: relative;
```

File: .changeset/metal-swans-allow.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+`npm install -g jscodeshift`
+
+Then, run the codemod using the appropriate command (examples given from the primer/components root directory)
+
+typescript example:
+`jscodeshift -t codemods/deprecateUtilityComponents.js --parser=tsx path/to/workspace/src/*.tsx`
+
+babel example:
+`jscodeshift -t codemods/deprecateUtilityComponents.js --parser=babel path/to/workspace/src/*.jsx`
```
Comment (colebemis) [inline .changeset/metal-swans-allow.md:30]: Let's use code blocks (```) for the code instructions:

````suggestion

```shell
npm install -g jscodeshift
```

Then, run the codemod using the appropriate command (examples given from the primer/components root directory)

TypeScript example:

```shell
jscodeshift -t codemods/deprecateUtilityComponents.js --parser=tsx path/to/workspace/src/*.tsx
```

Babel example:
```shell
jscodeshift -t codemods/deprecateUtilityComponents.js --parser=babel path/to/workspace/src/*.jsx
```
````
```

File: .changeset/metal-swans-allow.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+The following components will be deprecated:
+
+Flex
+Grid
+Position
+Absolute
+Fixed
+Relative
+Sticky
+BorderBox
```
Comment (colebemis) [inline .changeset/metal-swans-allow.md:16]: Let's use a list here:
```suggestion
- Flex
- Grid
- Position
- Absolute
- Fixed
- Relative
- Sticky
- BorderBox
```

**Pull request merged**